null_safety/calculate_lix/bin/main.dart

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2020, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:calculate_lix/lix.dart' as lix;
+
+void main(List<String> arguments) {
+  // Parse the arguments; we expect a single argument containing the file name.
+  if (arguments.length != 1) {
+    printUsage();
+    exit(64);
+  } else {
+    // Get the file name.
+    //
+    // TIP: We're using null safe versions of the core libraries, and we already
+    // checked that there is a single argument, so we can safely assume the
+    // first argument isn't null.
+    final fileName = arguments[0];
+    print("Calculating Lix of '$fileName'");
+
+    // Calculate lix.
+    try {
+      final lc = lix.calculate(File(fileName).readAsStringSync());
+      print("Lix is: ${lc.readability}, ${lc.describe} to read ("
+          "words: ${lc.words}, long words: ${lc.longWords}, "
+          "periods: ${lc.periods}).");
+    } catch (Exception) {
```

Comment:
You must make sure you throw `ArgumentError`, of course.

Reply:
We should really have a lint for that `(Exception)` thing.

Reply:
...I think we do!

Reply:
@filiph - good catch!

@pq recently improved the lint for this; Phil, what's the lint to enable for this repo? We should also add it to our recommended set of dart lints.

Reply:
Oops, I started making the change to use an exception rather than return -1, but forgot to finish it :-(

I look forward to `pub run test` working with Null safety!

Reply:
[`avoid_types_as_parameter_names`](https://dart-lang.github.io/linter/lints/avoid_types_as_parameter_names.html) 👍